src/wutil.cpp

```diff
@@ -564,7 +564,7 @@ file_id_t file_id_t::file_id_from_stat(const struct stat *buf)
 #if defined(__APPLE__) || defined(__DragonFly__) || defined(__FreeBSD__) ||  defined(__OpenBSD__) || defined(__NetBSD__)
     result.generation = buf->st_gen;
 #else
-    result.generation = 0;
+    result.generation = NULL;
```

Comment:
The document you referred to begins with

> The macro NULL is an implementation-defined null pointer constant

Notice the word "pointer" in that sentence. It is not idiomatic to use NULL in a non-pointer context regardless of the fact that in practice, today, it will always evaluate to zero.